### PR TITLE
[#1581] add missing fields in service

### LINKED ISSUE #1581: Add new missing fields in service model
Add fields:
- [x] resource_geographic_locations - multiselect (Country)
- [x] certifications - multiselect
- [x] standards - multiselect
- [x] open_source_technologies - multiselect
- [x] changelog - multiselect
- [x] grant_project_names - multiselect
- [x] last_update - date

Add associations:
- [x] access_type - multiselect (vocabulary)
- [x] access_mode - multiselect (vocabulary)

Other
- [x] Add new fields to `jms` and `eic:import`
- [x] Add new fields to tests

### PR DESCRIPTION
**Pull request by goreck888**
Add new fields in service model:
- resource_geographic_locations - multiselect (Country)
- certifications - multiselect
- standards - multiselect
- open_source_technologies - multiselect
- changelog - multiselect
- grant_project_names - multiselect
- last_update - date

Add new associations to service:
- access_type - multiselect (vocabulary)
- access_mode - multiselect (vocabulary)

Fixes #1581

### CODE REVIEW

**Comment by goreck888** [comment]
> Add also this to import and jms and test them
> 
> sory I did not notice

I have to add jms and eic import to the new fields I added today, but thanks for notice.

**Comment by agpul** [comment]
@wziajka @goreck888 I can't reach this PR

File: app/policies/backoffice/service_policy.rb [hunk 1]
```diff
@@ -95,16 +95,21 @@ def permitted_attributes
...
       :order_target, :status_monitoring_url, :maintenance_url,
       :order_url, :payment_model_url, :pricing_url,
       [funding_body_ids: []], [funding_program_ids: []],
+      [access_type_ids: []], [access_mode_ids: []],
+      [certifications: []], [standards: []],
+      [grant_project_names: []], [open_source_technologies: []],
+      [changelog: []], [multimedia: []],
       :activate_message, :logo, [trl_ids: []],
-      [contact_emails: []], [scientific_domain_ids: []],
+      :contact_emails, [scientific_domain_ids: []],
```
Comment (martaswiatkowska) [inline app/policies/backoffice/service_policy.rb:112]: can you add this change to proper PR-? and other connected?

File: spec/fixtures/files/eic_import_output.json [hunk 2]
```diff
@@ -44,7 +75,7 @@
     "userManual": "https://github.com/phnmnl/phenomenal-h2020/wiki",
     "trainingInformation": "http://phenomenal-h2020.eu/home/training-online",
     "feedback": "http://phenomenal-h2020.eu/home/help",
-    "price": "http://phenomenal-h2020.eu/home/wp-content/uploads/2016/09/Phenomenal-Terms-of-Use-version-11.pdf",
+    "accessPolicy": "http://phenomenal-h2020.eu/home/wp-content/uploads/2016/09/Phenomenal-Terms-of-Use-version-11.pdf",
```
Comment (martaswiatkowska) [inline spec/fixtures/files/eic_import_output.json:78]: EIC doesn't send  that field

File: app/policies/backoffice/service_policy.rb (hunk 1 above)
Comment (goreck888) [reply app/policies/backoffice/service_policy.rb:112]: In other connected PR this field is removed, I changed it only for test passing, but when I merge contact fields, this one change will be removed.

File: app/models/service.rb [hunk 3]
```diff
@@ -129,9 +133,11 @@ class Service < ApplicationRecord
   validates :manual_url, mp_url: true, if: :manual_url?
   validates :helpdesk_url, mp_url: true, if: :helpdesk_url?
   validates :helpdesk_email, allow_blank: true, email: true
-  validates :contact_emails, array: { email: true }
+  validates :use_cases_url, mp_url: true, if: :use_cases_url?
+  validates :privacy_policy_url, mp_url: true, if: :privacy_policy_url?
   validates :training_information_url, mp_url: true, if: :training_information_url?
   validates :language_availability, array: true
+  validates :resource_geographic_locations, array: true
```
[the remark this answers is not in the document]
Comment (goreck888) [reply app/models/service.rb:140]: Ok, I'll remove that.

File: app/javascript/app/controllers/service_controller.js [hunk 4]
```diff
@@ -2,27 +2,30 @@ import { Controller } from "stimulus"
...
-    var lastEmailField = document.createElement("input")
-    var parent = this.contactEmailsTarget
-    lastEmailField.type = "email"
-    lastEmailField.name = "service[contact_emails][]"
-    lastEmailField.id = "service_contact_emails_"+(parent.children.length - 1)
+    const lastArrayField = document.createElement("textarea")
+    const parent = this.arrayTarget
 
-    parent.appendChild(lastEmailField)
+    console.log(this.inputTarget)
```
Comment (martaswiatkowska) [inline app/javascript/app/controllers/service_controller.js:15]: You can remove this

Comment (goreck888) [reply app/javascript/app/controllers/service_controller.js]: Ok

**Pull request merged**